### PR TITLE
[script.keymap] 1.1.2

### PR DESCRIPTION
**Pull request by pkscout**
### Description
- Better support for v18 Leia
- Fix so script only backups user defined keymaps if trying to save new keymap file
- Moved all text into language file for easier internationalization

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.keymap/addon.xml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+v1.1.2
+- Better support for v18 Leia
+- Fix so script only backups user defined keymaps if trying to save new keymap file
+- Moved all text into language file for easier internationalization
+		</news>
+		<assets>
+        	<icon>icon.png</icon>
+        </assets>
+		<summary lang="en_GB">An editor for keymap files</summary>
+		<description lang="en_GB">Keymap Editor is a GUI for configuring mappings for remotes, keyboard and other inputs supported by xbmc. Note: existing user defined key mappings will be renamed to .bak.</description>
```
Comment (enen92) [inline script.keymap/addon.xml:20]: Kodi

File: script.keymap/addon.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		<platform>all</platform>
+		<license>GPL-3.0-only</license>
+		<forum>https://forum.kodi.tv/showthread.php?tid=168767</forum>
+		<source>https://github.com/tamland/xbmc-keymap-editor</source>
```
Comment (enen92) [inline script.keymap/addon.xml:24]: Please change upstream location

File: script.keymap/resources/lib/keymap.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+
+def main():
+    ## load mappings ##
+    try:
+        setup_keymap_folder()
+    except Exception:
+        traceback.print_exc()
+        utils.rpc('GUI.ShowNotification', title="Keymap Editor",
+            message="Failed to remove old keymap file", image='error')
```
Comment (enen92) [inline script.keymap/resources/lib/keymap.py:46]: Localise

File: script.keymap/resources/lib/keymap.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            utils.write_keymap(userkeymap, gen_file)
+            xbmc.executebuiltin("action(reloadkeymaps)")
+            break
+        elif idx == -1 and confirm_discard:
+            if Dialog().yesno(tr(30000), tr(30006)) == 1:
+                break
+        else:
+            break
+
+    sys.modules.clear()
```
Comment (enen92) [inline script.keymap/resources/lib/keymap.py:97]: Why is this needed?

File: script.keymap/resources/lib/editor.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+class Editor(object):
+    def __init__(self, defaultkeymap, userkeymap):
+        """Create the editor object."""
+        self.defaultkeymap = defaultkeymap
+        self.userkeymap = userkeymap
+        self.dirty = False
+
+    def start(self):
+        while True:
```
Comment (enen92) [inline script.keymap/resources/lib/editor.py:34]: Forever running loops should check for abortion. I'd use xbmc.abortRequested flag for Leia and monitor.abortRequested() for matrix

**Review by enen92 — CHANGES_REQUESTED** [review]

File: script.keymap/resources/lib/keymap.py (hunk 4 above)
Comment (pkscout) [reply script.keymap/resources/lib/keymap.py:97]: Oh god, I have no idea. I was trying not to poke this thing to hard while updating it. Let me test without and see what happens.

File: script.keymap/addon.xml (hunk 1 above)
Comment (pkscout) [reply script.keymap/addon.xml]: fixed in 9d51e34

File: script.keymap/resources/lib/editor.py (hunk 5 above)
Comment (pkscout) [reply script.keymap/resources/lib/editor.py]: fixed in 1a6d491

File: script.keymap/resources/lib/keymap.py (hunk 4 above)
Comment (pkscout) [reply script.keymap/resources/lib/keymap.py]: fixed in e8e1e87

File: script.keymap/resources/lib/keymap.py (hunk 3 above)
Comment (pkscout) [reply script.keymap/resources/lib/keymap.py]: fixed in 6b2acc5

File: script.keymap/addon.xml (hunk 2 above)
Comment (pkscout) [reply script.keymap/addon.xml:24]: As per our Slack conversation, I'm ignoring this.  '-)

**Review by enen92 — APPROVED** [review]
